client/client.go

```diff
@@ -0,0 +1,130 @@
+package client
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/Layr-Labs/eigenda-proxy/common"
+	"github.com/Layr-Labs/eigenda-proxy/eigenda"
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+const (
+	// NOTE: this will need to be updated as plasma's implementation changes
+	decodingOffset = 2
+)
+
+// TODO: Add support for custom http client option
+type Config struct {
+	URL string
+}
+
+// ProxyClient is an interface for communicating with the EigenDA proxy server
+type ProxyClient interface {
+	Health() error
+	GetData(ctx context.Context, cert *common.Certificate, domain common.DomainType) ([]byte, error)
+	SetData(ctx context.Context, b []byte) (*common.Certificate, error)
+}
+
+// client is the implementation of ProxyClient
+type client struct {
+	cfg        *Config
+	httpClient *http.Client
+}
+
+func New(cfg *Config) ProxyClient {
+	return &client{
+		cfg,
+		http.DefaultClient,
+	}
+}
+
+// Health indicates if server is operational; useful for event based awaits
+// when integration testing
+func (c *client) Health() error {
+	url := c.cfg.URL + "/health"
+	req, err := http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		return err
+	}
+
+	resp, err := c.httpClient.Do(req)
+	if err != nil {
+		return err
+	}
+
+	if resp.StatusCode != http.StatusOK {
```

Comment:
Would be nice to add a protocol for including error messages from the response body in the error output here.